components/gateway/cmd/main.go

```diff
@@ -25,23 +23,11 @@ func main() {
 	err := envconfig.InitWithPrefix(&cfg, "APP")
 	exitOnError(err, "Error while loading app config")
 
-	directorUrl, err := url.Parse(cfg.DirectorOrigin)
-	exitOnError(err, "Error while parsing Director URL")
+	directorProxy, err := proxy.New(cfg.DirectorOrigin)
+	exitOnError(err, "Error while initializing proxy")
 
 	log.Printf("Proxying requests to Director: %s\n", cfg.DirectorOrigin)
 
-	directorProxy := httputil.ReverseProxy{
-		Director: func(r *http.Request) {
-			r.URL.Scheme = directorUrl.Scheme
-			r.URL.Host = directorUrl.Host
-			r.URL.Path = singleJoiningSlash(directorUrl.Path, r.URL.Path)
-			r.Header = http.Header{
-				"Content-Type": []string{"application/json"},
-			}
-			r.Host = directorUrl.Host
-		},
-	}
-
 	router := mux.NewRouter()
 	router.HandleFunc("/", directorProxy.ServeHTTP)        // GraphQL Playground
```

Comment:
Isn't it enough to only handle requests on `/` as the handler function is the same?

Reply:
Hmm, I was thinking about forwarding requests to Connector and Healthchecker - that's why I specified exact paths to support future cases.